src/Hamiltonians/HOCartesian.jl

```diff
@@ -0,0 +1,417 @@
+"""
+    four_oscillator_integral_general(i, j, k, l; max_level = typemax(Int))
+
+Integral of four one-dimensional harmonic oscillator functions, following Titchmarsh (1948).
+State indices `i,j,k,l` start at `0` for the groundstate.
+"""
+function four_oscillator_integral_general(i, j, k, l; max_level = typemax(Int))
+    all(0 .≤ (i, j, k, l) .≤ max_level) || return 0.0
+    iseven(i + j + k + l) || return 0.0
+
+    # enforce correct symmetry and k ≥ l
+    i, j, k, l = sort(SVector(i, j, k, l); rev = true)
+    
+    a = i + j - k + l + 1    
+    b = i - j + k - l + 1
+    c = -i + j + k - l + 1
+    d = -i - j + k - l + 1
+    
+    p = sqrt(2 * gamma(i + 1) * gamma(j + 1) * gamma(k + 1) * gamma(l + 1)) * pi^2
+    q = gamma(a/2) * gamma(b/2) * gamma(c/2) * gamma(k + 1) / gamma(k - l + 1)
+    
+    f = _₃F₂(float(-l), b/2, c/2, float(1 + k - l), d/2, 1.0)
+    
+    return f * q / p
+end
+
+"""
+    largest_two_point_interval(i, j, [v=1,] w) -> range
+
+For two points `i` and `j` on a range `v:w`, find the largest subinterval of sites 
+such that moving `i` to one of those sites and moving `j` by an equal but opposite
+amount leaves both within `v:w`.
+
+`i` and `j` may be `Rational`s, but `v` and `w` must be `Integer`s.
+"""
+function largest_two_point_interval(i::Int, j::Int, v::Int, w::Int)
+    i in v:w && j in v:w || throw("interval out of bounds: i=$i, j=$j, v=$v, w=$w")
+    left_edge, right_edge = two_point_interval_bounds(i, j, v, w)
+    return left_edge:right_edge
+end
+function largest_two_point_interval(i, j, v::Int, w::Int)
+    v ≤ i ≤ w && v ≤ j ≤ w || throw("interval out of bounds: i=$i, j=$j, v=$v, w=$w")
+    left_edge, right_edge = two_point_interval_bounds(i, j, v, w)
+    return ceil(Int, left_edge):floor(Int, right_edge)
+end
+# largest_two_point_interval(i::Int, j::Rational, v, w) = largest_two_point_interval(i//1, j, v, w)   # are these necessary?
+# largest_two_point_interval(i::Rational, j::Int, v, w) = largest_two_point_interval(i, j//1, v, w)
+largest_two_point_interval(i, j, w) = largest_two_point_interval(i, j, 1, w)
+
+function two_point_interval_bounds(i, j, v, w)
+    if i ≤ j
+        left_gap = min(i - v, w - j)
+        left_edge = i - left_gap
+        right_gap = min(w - i, j - v)
+        right_edge = i + right_gap
+    else
+        left_gap = min(j - v, w - i)
+        left_edge = j - left_gap
+        right_gap = min(w - j, i - v)
+        right_edge = j + right_gap
+    end
+    return left_edge, right_edge
+end
+
+# useful conversion of aspect ratio to a factor for getting box size
+aspect_to_box(η::NTuple{N,Int}) where {N} = Int.(lcm(η...) .// η)
```

Comment:
```suggestion
aspect_to_box(η::NTuple{<:Any,Int}) = Int.(lcm(η...) .// η)
```
When not using the parameter, you can do this. I find that a bit easier to read, but you can do what you prefer.